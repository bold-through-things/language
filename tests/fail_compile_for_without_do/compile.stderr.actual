[
  {
    "recoverable": false,
    "line": 2,
    "char": 0,
    "content": "for i in",
    "error": "failed to assert: expected 'do' after 'for' but found 'then'",
    "error_type": "EXPECTED_DO_AFTER"
  },
  {
    "recoverable": false,
    "line": 2,
    "char": 0,
    "content": "for i in",
    "error": "failed to assert: must have a `do` block",
    "error_type": "ASSERTION_FAILED"
  },
  {
    "recoverable": false,
    "line": 10,
    "char": 0,
    "content": "67lang:call i",
    "error": "failed to assert: could not find a matching overload for i with arguments []",
    "error_type": "NO_MATCHING_OVERLOAD",
    "visible_overloads": [
      {
        "convention": {
          "convention": "FieldCall",
          "field": "i",
          "demands": [
            {
              "name": "HTMLElementTagNameMap",
              "type_params": [],
              "fields": []
            }
          ],
          "returns": {
            "name": "HTMLElement",
            "type_params": [],
            "fields": []
          }
        },
        "reasons": [
          {
            "reason_type": "ARITY_MISMATCH",
            "demanded_arity": 1,
            "received_arity": 0
          }
        ]
      },
      {
        "convention": {
          "convention": "FieldCall",
          "field": "i",
          "demands": [
            {
              "name": "HTMLElementTagNameMap",
              "type_params": [],
              "fields": []
            },
            {
              "name": "HTMLElement",
              "type_params": [],
              "fields": []
            }
          ],
          "returns": {
            "name": "None",
            "type_params": [],
            "fields": []
          }
        },
        "reasons": [
          {
            "reason_type": "ARITY_MISMATCH",
            "demanded_arity": 2,
            "received_arity": 0
          }
        ]
      }
    ]
  }
]
