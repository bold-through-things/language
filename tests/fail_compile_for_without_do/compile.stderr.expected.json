{
  "$items": [
    {
      "recoverable": false,
      "line": 2,
      "char": 0,
      "content": { "$contains": "for i in" },
      "error_type": "EXPECTED_DO_AFTER"
    },
    {
      "recoverable": false,
      "line": 2,
      "char": 0,
      "content": { "$contains": "for i in" },
      "error_type": "ASSERTION_FAILED"
    },
    {
      "recoverable": false,
      "line": 10,
      "char": 0,
      "content": { "$contains": "call i" },
      "error_type": "NO_MATCHING_OVERLOAD",
      "visible_overloads": { "$len": { "$ge": 1 } }
    }
  ],
  "$order": "any"
}
